pkg/gitprovider/aws-codecommit.go

```diff
@@ -0,0 +1,323 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitprovider
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/codecommit"
+	"github.com/aws/aws-sdk-go-v2/service/codecommit/types"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go/aws"
+	log "github.com/sirupsen/logrus"
+)
+
+type AwsCodeCommitGitProvider struct {
+	*AbstractGitProvider
+	baseApiUrl string
+	region     string
+}
+
+func NewAwsCodeCommitGitProvider(baseApiUrl string) *AwsCodeCommitGitProvider {
+	region := strings.TrimPrefix(strings.Split(baseApiUrl, ".")[0], "https://")
+	gitProvider := &AwsCodeCommitGitProvider{
+		AbstractGitProvider: &AbstractGitProvider{},
+		baseApiUrl:          baseApiUrl,
+		region:              region,
+	}
+	gitProvider.AbstractGitProvider.GitProvider = gitProvider
+
+	return gitProvider
+}
+
+func (g *AwsCodeCommitGitProvider) GetNamespaces() ([]*GitNamespace, error) {
+	// AWS CodeCommit does not have a project and repository structure similar to other git providers.
+	// Therefore, returning an repositories as array of type GitNamespace.
```

Comment:
please change to "Therefore, returning repositories as an array of type GitNamespace."